app/workers/facilities/access_data_download.rb

```
@@ -0,0 +1,125 @@
+      'CARDIOLOGY' => 'cardiology',
+      'GASTROENTEROLOGY' => 'gastroenterology',
+      'OPHTHALMOLOGY' => 'opthalmology',
+      'OPTOMETRY' => 'optometry',
+      'UROLOGY CLINIC' => 'urology_clinic'
+    }.freeze
+
+    WT_REQUIRED_KEYS = %w(facilityID ApptTypeName newWaitTime estWaitTime sliceEndDate).freeze
+
+    def update_cache(model, facilities)
+      facilities.each do |k, v|
+        attrs = { station_number: k,
+                  metrics: v['metrics'],
+                  source_updated: v['source_date'],
+                  local_updated: Time.now.utc.iso8601 }
+        obj = model.find(k)
+        if obj
+          obj.update(attrs)
+        else
+          model.create(attrs)
+        end
+      end
+    end
+
+    def invalidate_removed(model, facility_keys)
+      invalidate = model.keys - facility_keys
+      invalidate.each { |x| model.delete(x) }
+    end
+
+    def require_keys(record, required_keys)
+      diff = required_keys - record.keys
+      raise AccessDataError, "Missing expected keys: #{diff}" unless diff.blank?
+    end
+
+    def parse_satisfaction_data(records)
+      facilities = Hash.new { |h, k| h[k] = { 'metrics' => {} } }
+      records.each do |rec|
+        require_keys(rec, SAT_REQUIRED_KEYS)
+        id = rec['facilityID']
+        facility = facilities[id]
+        category = SAT_KEY_MAP[rec['ApptTypeName']]
+        facility['metrics'][category] = rec['SHEPScore']
+        facility['source_date'] = rec['sliceEndDate']
+      end
+      facilities
+    end
+
+    def update_satisfaction_data(client)
+      records = client.download
+      facilities = parse_satisfaction_data(records)
+      update_cache(FacilitySatisfaction, facilities)
+      invalidate_removed(FacilitySatisfaction, facilities.keys)
+    rescue Common::Exceptions::BackendServiceException, Common::Client::Errors::ClientError => e
+      log_exception_to_sentry(e)
+    rescue Facilities::AccessDataError => e
+      log_exception_to_sentry(e)
+    end
```

Don't you want to re-raise the exception to halt execution?